src/App.js

```diff
@@ -14,11 +14,103 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: ['Jam'],
+  }
+
+  addProduct = (product) => {
+    this.setState(prevState => ({
+      selectedGoods: prevState.selectedGoods.concat(product),
+    }));
+  }
+
+  removeProduct = (good) => {
+    this.setState(prevState => ({
+      selectedGoods:
+        prevState.selectedGoods.filter(product => product !== good),
+    }));
+  }
+
+  clearProductList = () => {
+    this.setState({ selectedGoods: [] });
+  }
+
+  createProductList = () => {
+    if (this.state.selectedGoods.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (this.state.selectedGoods.length < 3) {
+      return this.state.selectedGoods.join(' and ');
+    }
+
+    return `${this.state.selectedGoods.slice(0, -2).join(', ')
+    }, ${this.state.selectedGoods.slice(-2).join(' and ')}`;
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h1>
+          Selected good:
+          {this.createProductList()}
+        </h1>
+        {!!this.state.selectedGoods.length
+          && (
+            <button
+              type="button"
+              className="deleteButton"
+              onClick={() => this.clearProductList()}
+            >
+              Clear all
+            </button>
+          )}
+        <ul className="app__list">
+          {goodsFromServer.map((good) => {
+            const isProductSelected = this.state.selectedGoods.includes(good);
+
+            return (
+              <div key={good}>
```

Comment:
Можна також додати фрагмент замість div
```suggestion
              <React.Fragment key={good}>
```